### ISSUE TITLE
Incorrect Change History Marker

### ISSUE THREAD

**Issue by curya** [notepad-plus-plus/notepad-plus-plus#12283]
### Description of the Issue
Incorrect 'Change History' marker when working with an already saved file. Issue does not happen for an unsaved file.

### Steps to Reproduce the Issue
1. Open already saved file.
2. Type something on a new line. Let's say line 10. Press Enter.
3. Line 10 will be orange, but line 11 will be green.

### Expected Behavior
Both line 10 and line 11 should be orange.

### Actual Behavior
Line 10 is orange but line 11 is green.

### Other info
Although this doesn't happen for an unsaved file, there is a separate issue that *does* happen on an unsaved file. If you type something on line 1, it will be orange. But if you press enter, line 2 will have no color until you start typing. Not sure if that is intended.

**Comment by alankilborn** [comment]
Cannot reproduce green on line 11:

![image](https://user-images.githubusercontent.com/77065706/193898804-85cfbb4e-5fd9-4684-8e03-b7d353235057.png)

Also, it arguable that line 11 should or should not be colored.

Notepad++ v8.4.6   (64-bit)
Build time : Sep 25 2022 - 19:51:39
Path : C:\\........\npp.8.4.6.portable.x64\notepad++.exe
Command Line : -titleAdd="8.4.6 64-bit" -multiInst
Admin mode : OFF
Local Conf mode : ON
Cloud Config : OFF
OS Name : Windows 10 Enterprise (64-bit) 
OS Version : 21H2
OS Build : 19044.1889
Current ANSI codepage : 1252
Plugins : 
    mimeTools (2.8)
    NppConverter (4.4)
    NppExport (0.4)

**Comment by HydroHydro** [comment]
I can do it every single time.
![image](https://user-images.githubusercontent.com/43591297/193941931-07e54400-7888-4e4b-8847-7b5173c32aaa.png)
 Did you save the file first and THEN do the line 11 thing? It only happens on a file that has already been saved.

And you're right. I was wondering if I should mention that. Line 11 probably shouldn't be colored at all.

**Comment by curya** [comment]
I can do it every single time.
![image](https://user-images.githubusercontent.com/43591297/193941931-07e54400-7888-4e4b-8847-7b5173c32aaa.png)
 Did you save the file first and THEN do the line 11 thing? It only happens on a file that has already been saved.

And you're right. I was wondering if I should mention that. Line 11 probably shouldn't be colored at all.

**Comment by alankilborn** [comment]
>Did you save the file first and THEN do the line 11 thing?

Yes, started with a saved file with initially no color at all in the change margin.

>Apparently this issue also only happens at the end of a text file.

I couldn't replicate that, either. :-(

**Comment by ArkadiuszMichalski** [comment]
Something is wrong when we make change and press `Enter`, STR:
1. Open some empty file
2. Put in first line 'aaa' << orange marker appear for 1st line
3. Press `Enter` << notice that we don't have the orange marker for 2nd line
4. Press `Backspace`  and once again press `Enter` << now orange marker appear 2nd line
5. Press `Backspace` and save file << green marker appear for 1st line
6. Press `Enter` << 1st line has orange marker and 2nd line has green marker
 
I haven't used this change history yet so I don't know if it's a bug or not.

**Comment by alankilborn** [comment]
@ArkadiuszMichalski 
I can reproduce your results following your STR.
Definitely some inconsistencies (bugs) here; but I wouldn't say they are Notepad++ bugs.
I'm sure some evolution is coming in the change history feature in Scintilla; Notepad++ will have to wait for that.

**Comment by Baptistou** [comment]
I would like to provide additional information about @ArkadiuszMichalski test case.
When switching to Unix line return and displaying all characters, I noticed that "step 3 Press `Enter`" does not add `LF` at the end of line after `CRLF` but inserts `LF` before `CRLF` which indeed modifies the first line instead of the second line.
See screenshots below.

![image](https://user-images.githubusercontent.com/16428197/200359003-18a470fb-e5af-4772-b7cb-4fa60bdb5745.png)
![image](https://user-images.githubusercontent.com/16428197/200359043-c1b3c93b-10f8-4a1f-9d6e-7dabeb7bb26e.png)

**Comment by alankilborn** [comment]
>I would like to provide additional information about @ArkadiuszMichalski test case.

>When switching to Unix line return

How are you doing that?
Please provide step-by-step as @ArkadiuszMichalski did, so others can attempt to replicate.

**Comment by Baptistou** [comment]
> How are you doing that "switch"?
> Please provide step-by-step as @ArkadiuszMichalski did, so others can attempt to replicate.
> Your screenshot shows you are obtaining mixed line-endings in your file, which goes "beyond" the issue under discussion here.

1. Open a new file with Windows Format (CRLF)
2. Go to menu Edit > EOL Conversion > UNIX Format (LF)
3. Undo the file modifications, menu Edit > Undo or `CTRL+Z`
4. Add a new line to the file

**Comment by alankilborn** [comment]
Start here:

![image](https://user-images.githubusercontent.com/77065706/200421909-ba22b43b-548d-450f-a7cf-e10f4baaf375.png)

Then convert line endings:

![image](https://user-images.githubusercontent.com/77065706/200421986-fbdd1510-7475-4a78-bdb9-805a526bf207.png)

Then, after Undo (note no visible caret):

![image](https://user-images.githubusercontent.com/77065706/200422119-48a28e54-4eee-4255-9a64-ebfbe1196187.png)

Then, after an Enter press:

![image](https://user-images.githubusercontent.com/77065706/200422285-48de8906-4cdc-4742-8568-ee6003f88c93.png)

This could be a separate problem, but it is not related to the topic of this current issue (change history).

**Comment by Baptistou** [comment]
> This could be a separate problem, but it is not related to the topic of this current issue (change history).

OK I created a separate issue #12472
